File: Code/SimpleBudget.API/Models/PlanPayments/PlanPaymentEditItemModel.cs
using System.ComponentModel.DataAnnotations;

namespace SimpleBudget.API.Models
{
    public class PlanPaymentEditItemModel
    {
        public bool IsActive { get; set; }

        [Required]
        public string PaymentType { get; set; } = default!;

        [Required]
        public string StartDate { get; set; } = default!;

        public string? EndDate { get; set; }

        public string? Company { get; set; }

        public string? Category { get; set; }

        [Required]
        public string Wallet { get; set; } = default!;

        public string? Description { get; set; }

        [Required]
        public decimal Value { get; set; }

        public bool Taxable { get; set; }

        public int? TaxYear { get; set; }

        public string? Person { get; set; }
    }
}

File: Code/client/src/api/models/PlanPaymentEditItemModel.ts
export interface PlanPaymentEditItemModel {
    isActive: boolean;
    paymentType: string;
    startDate: string;
    endDate: string | undefined | null;
    company: string | undefined | null;
    category: string | undefined | null;
    wallet: string;
    description: string | undefined | null;
    value: number;
    taxable: boolean;
    taxYear: number | undefined | null;
    person: string | undefined | null;
}